### PR TITLE
Add the waiter for ec2_vpc_nat_gateway

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Add waiters for ec2_vpc_nat_gateway (https://github.com/ansible-collections/community.aws/pull/445) to manage deleted and available states.

### CODE REVIEW

**Review by tremble — CHANGES_REQUESTED** [review]

File: plugins/module_utils/waiters.py [hunk 1]
```diff
@@ -299,6 +299,24 @@
                 },
             ]
         },
+         "NatGatewayDeleted": {
```
[the remark this answers is not in the document]
Comment (alinabuzachis) [reply plugins/module_utils/waiters.py]: Hi @tremble, thank you for reviewing. I corrected the indentation and added the waiter needed to manage the Nat Gateway available state.

**Comment by alinabuzachis** [comment]
recheck

File: plugins/module_utils/waiters.py [hunk 2]
```diff
@@ -299,6 +299,42 @@
...
+            "operation": "DescribeNatGateways",
+            "acceptors": [
+                {
+                    "state": "success",
+                    "matcher": "pathAll",
+                    "expected": "available",
+                    "argument": "NatGateways[].State"
+                },
+                {
+                    "state": "success",
```
Comment (s-hertel) [inline plugins/module_utils/waiters.py:332]: it should be "retry" here, I think

Comment (alinabuzachis) [reply plugins/module_utils/waiters.py]: @s-hertel You are right! Just changed! When copying and pasting I forget to change. Thank you.

**Review by tremble — APPROVED** [review]

**Pull request merged**